docs/buildah-bud.md

```diff
@@ -622,7 +631,7 @@ Only the current container can use a private volume.
 
   Note:
 
-     - The `O` flag is not allowed to be specified with the `Z` or `z` flags. Content mounted into the container is labeled with the private label.
+     - The `O` flag is not allowed to be specified with the `Z`, `z` or `U` flags. Content mounted into the container is labeled with the private label.
```

Comment:
Is there a reason to not allow Overlay options with the U Flag?  In SELinux the kernel will prevent this.  Might not be a good idea because it will trigger a copy-up, but it might be the case where content is intended to be used for the run of the container temporarily but not written to the host.  Of course this would mean the chown would need to happen on each run of the container.

Reply:
Thanks. Let me work on this.

Reply:
Currently buildah will chown the source volume and then use the processUID and processGID to create the overlay temp dir, would this be the expected behaviour? Initially I tried to chown the overlay temp directory only, in the case of rootless containers the UID/GID were added correctly, but in the case of rootful the content of the source volume was never copied up to the overlay temp dir so the correct UID/GID were never added to sub dirs/files.

Reply:
If you chown the files after the volume is mounted, it should create new INODES in the upper directory with the correct ownership.

Reply:
It seems to work with rootless containers since the volume is mounted using `fuse-overlayfs`  here https://github.com/containers/buildah/blob/master/pkg/overlay/overlay.go#L64, but for containers created by root the overlay volume is mounted by crun using the native overlay type.

If I am not wrong there's no way to specify uid/gid as an option for overlay native mount type. The only way I can see this working is if the real source volume is chown just like volumes without the O flag.